File: tricore-server/src/routes/sales.rs
use actix_web::web;
use crate::handlers::sales;

pub fn configure(cfg: &mut web::ServiceConfig) {
    cfg.service(
        web::scope("/sales")
            // Auth
            .route("/auth/login", web::post().to(sales::login))
            // Users
            .route("/users", web::get().to(sales::list_users))
            .route("/users/{id}", web::get().to(sales::get_user))
            .route("/users", web::post().to(sales::create_user))
            .route("/users/{id}", web::put().to(sales::update_user))
            // Products
            .route("/products", web::get().to(sales::list_products))
            .route("/products/{id}", web::get().to(sales::get_product))
            .route("/products", web::post().to(sales::create_product))
            .route("/products/{id}", web::put().to(sales::update_product))
            // Orders
            .route("/orders", web::get().to(sales::list_orders))
            .route("/orders/{id}", web::get().to(sales::get_order))
            .route("/orders", web::post().to(sales::create_order))
            .route("/orders/{id}", web::put().to(sales::update_order))
            .route("/orders/{id}/status", web::patch().to(sales::update_order_status))
            // Bundle Sales
            .route("/bundles", web::get().to(sales::list_bundles))
            .route("/bundles/{id}", web::get().to(sales::get_bundle))
            .route("/bundles", web::post().to(sales::create_bundle))
            // Returns
            .route("/returns", web::get().to(sales::list_returns))
            .route("/returns/{id}", web::get().to(sales::get_return))
            .route("/returns", web::post().to(sales::create_return))
            .route("/returns/{id}/status", web::patch().to(sales::update_return_status))
            // Dashboard
            .route("/dashboard", web::get().to(sales::dashboard)),
    );
}

File: tricore-web-rust/src/services/sales.ts
import api from './http'

export const salesAPI = {
  login: (data: { username: string; password: string }) =>
    api.post('/sales/auth/login', data),
  listUsers: (params?: Record<string, unknown>) => api.get('/sales/users', { params }),
  getUser: (id: string) => api.get(`/sales/users/${id}`),
  createUser: (data: Record<string, unknown>) => api.post('/sales/users', data),
  updateUser: (id: string, data: Record<string, unknown>) => api.put(`/sales/users/${id}`, data),
  listProducts: (params?: Record<string, unknown>) => api.get('/sales/products', { params }),
  getProduct: (id: string) => api.get(`/sales/products/${id}`),
  createProduct: (data: Record<string, unknown>) => api.post('/sales/products', data),
  updateProduct: (id: string, data: Record<string, unknown>) => api.put(`/sales/products/${id}`, data),
  listOrders: (params?: Record<string, unknown>) => api.get('/sales/orders', { params }),
  getOrder: (id: string) => api.get(`/sales/orders/${id}`),
  createOrder: (data: Record<string, unknown>) => api.post('/sales/orders', data),
  updateOrder: (id: string, data: Record<string, unknown>) => api.put(`/sales/orders/${id}`, data),
  updateOrderStatus: (id: string, data: Record<string, unknown>) => api.patch(`/sales/orders/${id}/status`, data),
  listBundles: (params?: Record<string, unknown>) => api.get('/sales/bundles', { params }),
  getBundle: (id: string) => api.get(`/sales/bundles/${id}`),
  createBundle: (data: Record<string, unknown>) => api.post('/sales/bundles', data),
  listReturns: (params?: Record<string, unknown>) => api.get('/sales/returns', { params }),
  getReturn: (id: string) => api.get(`/sales/returns/${id}`),
  createReturn: (data: Record<string, unknown>) => api.post('/sales/returns', data),
  updateReturnStatus: (id: string, status: string) => api.patch(`/sales/returns/${id}/status`, { status }),
  getDashboard: () => api.get('/sales/dashboard'),
  // Categories
  listCategories: () => api.get('/sales/categories'),
  getCategory: (id: string) => api.get(`/sales/categories/${id}`),
  createCategory: (data: Record<string, unknown>) => api.post('/sales/categories', data),
  updateCategory: (id: string, data: Record<string, unknown>) => api.put(`/sales/categories/${id}`, data),
  deleteCategory: (id: string) => api.delete(`/sales/categories/${id}`),
}
